src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -2373,46 +2362,60 @@ class StubGenerator: public StubCodeGenerator {
              tmpU = isLU ? tmp5 : tmp1, // where to keep U for comparison
              tmpL = isLU ? tmp1 : tmp5; // where to keep L for comparison
 
```

Comment:
I see `cnt1`(x12) is unused now. We could make use of this to further eliminate use of both `tmp4` (x7) and `tmp5` (x31).
Let:
```
             tmpU = isLU ? tmp2 : tmp1, // where to keep U for comparison
             tmpL = isLU ? tmp1 : tmp2; // where to keep L for comparison
```
And use `cnt1` and `tmp2` instead where `tmp4` and `tmp5` are used respectively. This would help save one `mv` and saving & restoring of `tmp4` and `tmp5`. Also remember to change `tmpLval` from `tmp4` (x7) to `cnt1`(x12) in compare_string_8_x_LU.

Reply:
It works, no difference in performance tho ( UL and LU cases)